File: kotlin/duck/Duck.kt
open class Duck(var name: String) {

    open fun quack() {
        println("$name is quacking")
    }

    fun swim() {
        println("$name is swimming")
    }

    open fun fly() {
        println("$name is flying")
    }

    fun display() {
        println("I am a $name")
    }

}

File: javascript/duck/Duck.js

export class Duck {
    constructor(name) { this.name = name }

    quack() {
        console.log(this.name + ' is quacking')
    }

    swim() {
        console.log(this.name + ' is swimming')
    }

    display() {
        console.log('I am a ' + this.name)
    }
}